packages/form-js-viewer/src/render/components/form-fields/FilePicker.js

```diff
@@ -40,21 +40,29 @@ export function FilePicker(props) {
   const evaluatedMultiple =
     useSingleLineTemplateEvaluation(typeof multiple === 'string' ? multiple : multiple.toString()) === 'true';
   const errorMessageId = `${domId}-error-message`;
-  const filesKey = `${id}_value_key`;
   const filesKey = `file::${calculateValuePath(props.fieldInstance.valuePath)}`;
+  const deleteFiles = useCallback(() => {
+    if (fileRegistry) {
+      fileRegistry.deleteFiles(filesKey);
+    }
+  }, [fileRegistry, filesKey]);
 
   useEffect(() => {
-    const reset = () => {
+    return () => {
```

Comment:
The problem with this is that by design, hiding operations are not destructive, so that if someone hides something by accident they don't lose their progress. So I think we'll have to go the route I initially suggested as otherwise we orphan file references and have different behavior for the file picker and regular components.

![image](https://github.com/user-attachments/assets/59b62908-a8e2-4bdc-af4b-7e9634b2068c)